multiview_prediction_toolkit/segmentation/ortho_segmentor.py

```diff
@@ -0,0 +1,315 @@
+
+    def create_sliding_window_dataset(
+        self, is_annotated: bool
+    ) -> SemanticSegmentationSlidingWindowGeoDataset:
+        """Sets self.dataset
+
+        Args:
+            is_annotated (bool): Should we use labels
+
+        Returns:
+            SemanticSegmentationSlidingWindowGeoDataset: The dataset, also sets self.dataset
+        """
+
+        # Keyword args shared between annotated and not
+        kwargs = {
+            "class_config": self.class_config,
+            "image_uri": self.raster_input_file,
+            "image_raster_source_kw": dict(allow_streaming=False),
+            "size": self.chip_size,
+        }
+
+        if is_annotated:
+            if self.raster_input_file is None and self.vector_label_file is None:
+                raise ValueError("One type of label must be included")
+
+            kwargs["label_vector_uri"] = self.vector_label_file
+            kwargs["label_raster_uri"] = self.raster_label_file
+            kwargs["stride"] = self.training_stride
+        else:
+            kwargs["stride"] = self.inference_stride
+
+        # Create the dataset
+        self.dataset = SemanticSegmentationSlidingWindowGeoDataset.from_uris(**kwargs)
+        return self.dataset
+
+    def write_training_chips(
+        self,
+        output_folder: PATH_TYPE,
+        brightness_multiplier: float = 1.0,
+        background_ind: int = 255,
+        skip_all_nodata_tiles: bool = True,
+    ):
+        """Write out training tiles from raster
+
+        Args:
+            output_folder (PATH_TYPE): The folder to write to, will create 'imgs' and 'anns' subfolders if not present
+            brightness_multiplier (float, optional): Multiply image brightness by this before saving. Defaults to 1.0.
+            background_ind (int, optional): Write this value for unset/nodata pixels. Defaults to 255.
+            skip_all_nodata_tiles (bool, optional): If a tile has no valid data, skip writing it
+        """
+        # Create annoated dataset
+        self.create_sliding_window_dataset(is_annotated=True)
+
+        num_labels = len(self.class_config.names)
+
+        # Create output folders
+        image_folder = Path(output_folder, "imgs")
+        anns_folder = Path(output_folder, "anns")
+
+        image_folder.mkdir(parents=True, exist_ok=True)
+        anns_folder.mkdir(parents=True, exist_ok=True)
+
+        # Main loop for writing out the data
+        for (image, label), window in tqdm(
+            zip(self.dataset, self.dataset.windows),
+            total=len(self.dataset),
+            desc="Saving train images and labels",
+        ):
+            label = label.cpu().numpy().astype(np.uint8)
+            mask = label == (num_labels - 1)
+
+            # If no data is valid, skip writing it
+            if skip_all_nodata_tiles and np.all(mask):
+                continue
+
+            # Set the nodata regions to the background value
+            label[mask] = background_ind
```

Comment:
Do we need to make sure the background value does not occur naturally as a (meaningful) label pixel value? I guess with the default of `255` this would probably only occur if there were that many classes (unlikely), so probably not a big deal? As long as the user is aware of this if/when they change `background_ind` from its default.

Reply:
Then we also need to keep track of which value is the mask. Why not keep it defined always as "the last label", as this method assumes it is to begin with (if I understand right)?